nats-base-client/headers.ts

```diff
@@ -182,41 +185,89 @@ export class MsgHdrsImpl implements MsgHdrs {
     return k.trim();
   }
 
-  get(k: string): string {
-    const key = MsgHdrsImpl.canonicalMIMEHeaderKey(k);
-    const a = this.headers.get(key);
-    return a ? a[0] : "";
+  keys(): string[] {
+    const keys = [];
+    for (const sk of this.headers.keys()) {
+      keys.push(sk);
+    }
+    return keys;
   }
 
-  has(k: string): boolean {
-    return this.get(k) !== "";
+  findKeys(k: string, match = Match.Exact): string[] {
+    const keys = this.keys();
+    switch (match) {
+      case Match.Exact:
+        return keys.filter((v) => {
+          return v === k;
+        });
+      case Match.CanonicalMIME:
+        k = canonicalMIMEHeaderKey(k);
+        return keys.filter((v) => {
+          return v === k;
+        });
+      default:
+        const lci = k.toLowerCase();
+        return keys.filter((v) => {
+          return lci === v.toLowerCase();
+        });
+    }
   }
 
-  set(k: string, v: string): void {
-    const key = MsgHdrsImpl.canonicalMIMEHeaderKey(k);
-    const value = MsgHdrsImpl.validHeaderValue(v);
-    this.headers.set(key, [value]);
+  get(k: string, match = Match.Exact): string {
+    const keys = this.findKeys(k, match);
+    if (keys.length) {
+      const v = this.headers.get(keys[0]);
+      if (v) {
+        return Array.isArray(v) ? v[0] : v;
+      }
+    }
+    return "";
+  }
+
+  has(k: string, match = Match.Exact): boolean {
+    return this.findKeys(k, match).length > 0;
+  }
+
+  set(k: string, v: string, match = Match.Exact): void {
+    this.delete(k, match);
+    if (match === Match.CanonicalMIME) {
```

Comment:
Since you rely on `append()`, which does that check, why do you do it here?

Reply:
Thank you good catch!